homeassistant/components/modbus/switch.py

```diff
@@ -134,7 +134,7 @@ async def async_turn_on(self, **kwargs):
         else:
             self._available = True
             if self._verify_active:
-                self.async_update()
+                await self.async_update()
```

Comment:
The climate and cover platforms do also need to be addressed.

Reply:
and switch, but I saved those 3 because they all not only read but also write.

If you prefer I can add them to this PR?

Reply:
Ups my mind was in the next PR, will change right now

Reply:
Ups, DONE